### ISSUE TITLE
Laggy Drag/Drop on Windows (dragging files from outside app into app)

### ISSUE THREAD

**Issue by JerContact** [superlistapp/super_native_extensions#265]
When dragging on windows into the app ( a file outside the app ), it feels laggy and not as responsive.  In the example video the red part is all a child of the drop region, and the drop region popup has 0 ms popup route.  This is also on release, seems laggy.

https://github.com/superlistapp/super_native_extensions/assets/38195772/3db86cf2-396f-4a33-b4d7-8f7461a0270f

**Comment by knopp** [comment]
There is a little bit of lag caused by underlying data object access, some of it can possibly be cached, but I tested it now and looks nowhere near as bad as the video. Is the video from debug or release build?

**Comment by JerContact** [comment]
Yeah, this is on release, i dragged in audio files, not sure if that adds to the slowness at all (.wav files that are about 80MB each), doing some test it does seem the files itself are slower than a folder (for instance)

**Comment by JerContact** [comment]
Here is a video showing the difference

https://github.com/superlistapp/super_native_extensions/assets/38195772/13198f82-7fe5-463c-b693-d24a572401bd

**Comment by knopp** [comment]
Do you access the reader when dropping the file over? Normally the file size should not matter at all - unless you're reading the file. Can you reproduce this with the included drag & drop example?

**Comment by JerContact** [comment]
Yeah, i tried running the example, but i just get a bunch of errors

```
Launching lib\main.dart on Windows in debug mode...
../../super_clipboard/lib/src/format_conversions.dart(85,15): error G44692867: A value of type 'List<int>' can't be returned from a function with return type 'Uint8List'. [C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\flutter\flutter_assemble.vcxproj]
../../super_clipboard/lib/src/format_conversions.dart(89,15): error G44692867: A value of type 'List<int>' can't be returned from a function with return type 'Uint8List'. [C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\flutter\flutter_assemble.vcxproj]
../../super_clipboard/lib/src/reader_internal.dart(221,45): error GC2F972A8: The argument type 'List<int>' can't be assigned to the parameter type 'Uint8List'. [C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\flutter\flutter_assemble.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(249,5): error MSB8066: Custom build for 'C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\CMakeFiles\909d59c0bfbc445efaf6e8ea72c2d6cb\flutter_windows.dll.rule;C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\CMakeFiles\23c7be1af70eec2290c8fdeae2107118\flutter_assemble.rule' exited with code 1. [C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\flutter\flutter_assemble.vcxproj]
Exception: Build process failed.


Exited (1).

```

**Comment by JerContact** [comment]
this is my DropRegion:

```
return DropRegion(
      formats: const [Formats.fileUri],
      onDropOver: (event) {
        return DropOperation.link;
      },
      onDropEnter: (event) {
        _hasDropped = false;
        showScanDragOverDialog(context,
            onDroppedConfirmed: () => _hasDropped = true);
      },
      onDropEnded: (event) {
        if (_hasDropped) return;
        Navigator.of(context).popUntil((route) {
          return route.isFirst;
        });
      },
      onPerformDrop: (_) async {
        return;
      },
```

**Comment by JerContact** [comment]
After some more tests, it's really the number of things you drag in that effects it.  These folder can be empty too

https://github.com/superlistapp/super_native_extensions/assets/38195772/1133fbee-a1f0-44cf-a2e2-fc97f7018530

**Comment by knopp** [comment]
> Yeah, i tried running the example, but i just get a bunch of errors
> 
> ```
> Launching lib\main.dart on Windows in debug mode...
> ../../super_clipboard/lib/src/format_conversions.dart(85,15): error G44692867: A value of type 'List<int>' can't be returned from a function with return type 'Uint8List'. [C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\flutter\flutter_assemble.vcxproj]
> ../../super_clipboard/lib/src/format_conversions.dart(89,15): error G44692867: A value of type 'List<int>' can't be returned from a function with return type 'Uint8List'. [C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\flutter\flutter_assemble.vcxproj]
> ../../super_clipboard/lib/src/reader_internal.dart(221,45): error GC2F972A8: The argument type 'List<int>' can't be assigned to the parameter type 'Uint8List'. [C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\flutter\flutter_assemble.vcxproj]
> C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(249,5): error MSB8066: Custom build for 'C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\CMakeFiles\909d59c0bfbc445efaf6e8ea72c2d6cb\flutter_windows.dll.rule;C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\CMakeFiles\23c7be1af70eec2290c8fdeae2107118\flutter_assemble.rule' exited with code 1. [C:\Users\Jeremy\Documents\super_native_extensions\super_drag_and_drop\example\build\windows\flutter\flutter_assemble.vcxproj]
> Exception: Build process failed.
> 
> 
> Exited (1).
> ```

This seems like you're trying to build with older flutter version?

**Comment by knopp** [comment]
If the number of files makes it slower https://github.com/superlistapp/super_native_extensions/pull/266 might improve that. Though there may be other parts where accessing the explorer DataObject is too slow.

I'm wondering if some explorer plugins could make this slower than it normally is.

**Comment by knopp** [comment]
Hi,

can you test `0.8.1` in release mode and let me know if the performance improves?

**Comment by JerContact** [comment]
Sadly, on my project, i can't update b/c of dependencies issues.  So, I tested on the example version.  This is what I see:

https://github.com/superlistapp/super_native_extensions/assets/38195772/44ba5c14-e019-4aee-9b8e-4d71a4d54958

**Comment by JerContact** [comment]
I'm not sure how to validate this with a new version, since this is what my pubspec.yaml looks like for this, and doesn't seem to do anything if I change it to 0.8.1 and do flutter pub upgrade

this is the pubspec.yaml file inside the "super_drag_and_drop" folder in the "example" folder

```
dependencies:
  flutter:
    sdk: flutter
  collection: ^1.16.0
  super_clipboard: ^0.8.0-dev.3
  super_drag_and_drop: ^0.8.0-dev.3
  super_clipboard_example:
    path: ../../super_clipboard/example
  super_context_menu: ^0.8.0-dev.3

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2

dev_dependencies:
  flutter_test:
    sdk: flutter
```

**Comment by knopp** [comment]
Did you pull the repository? 

As for your project, you should be able to force version using dependency overrides:

```yaml
dependency_overrides:
  super_drag_and_drop: ^0.8.1
  super_native_extensions: ^0.8.1
```

**Comment by JerContact** [comment]
Alrighty, i got 0.8.1 with the example project on release.  It's definitely better than it was, but still very slow for 100+ files, where I think users would think there is some bug going on

https://github.com/superlistapp/super_native_extensions/assets/38195772/cb4adfa6-9037-4ca3-9d88-48250a5e8e18

**Comment by knopp** [comment]
I have a work in progress branch that makes this much faster on windows. Here's drag & drop with over 4000 items:

https://github.com/superlistapp/super_native_extensions/assets/96958/86f003ee-fece-4c46-8426-31bdf5af3c96

It's will take some work to complete, but it's quite promising.

**Comment by JerContact** [comment]
Ohhhhhhhhh, that's very promising!!  Thanks for looking into this!

**Comment by knopp** [comment]
0.8.2 has been published. Please let me know if the performance improves for you.
